### DISCUSSION TITLE
Gitlocalize vs Crowdin

### DISCUSSION THREAD

**Discussion by mtdvlpr** [sircharlo/meeting-media-manager#995] [Polls]
I've been casually looking for Gitlocalize alternatives, because I have now experienced a number of shortcomings ([see my open issues](https://github.com/gitlocalize/feedback/issues/created_by/mtdvlpr)). I looked at Crowdin and Transifex.

Crowdin looks very promising for a number of reasons:
- You can translate both M³ and its documentation website from the same project
- Duplicate strings between M³ and its documentation will only have to be translated once (all setting labels exist in both)
- You can hide words that are not supposed to be translated (the ref property in the documentation pages)

Although these features are very nice, both platforms come with a big con: a much more complex UI. I don't think exposing our translators to such a complex UI is beneficial, but perhaps the translators feel different about it.

I have created [a test project on Crowdin](https://crowdin.com/project/test-project-m3/invite?h=57b484ede8f1943cfe25069e8d9975381551723) to test out the UI and translation process. Like Gitlocalize, you can create an account using your GitHub profile for convenience.

**Comment by joaochris** [comment]
@mtdvlpr In crowdin are you notified about new strings?

**Reply by mtdvlpr** [reply]
Yes! You should also be notified in Gitlocalize though.

**Reply by joaochris** [reply]
> Yes! You should also be notified in Gitlocalize though.

I never got a notification from him 😁

**Reply by mtdvlpr** [reply]
Then I guess that's another +1 for Crowdin :)

**Comment by sircharlo** [comment]
I think the problem with Crowdin was their free plan's limitations:
![image](https://user-images.githubusercontent.com/1275666/210414792-baf26301-08f5-45b7-994a-553d15857032.png)

**Reply by sircharlo** [reply]
Perhaps not an issue at the moment, but in the future if more languages and words are added (read: if features are added to the app), we might run into the word limit fairly quickly.

**Reply by mtdvlpr** [reply]
Ah, okay. You could try [registering as an open source project](https://crowdin.com/page/open-source-project-setup-request). I think we meet all the requirements. We'll have to setup the project on crowdin first.

**Comment by mtdvlpr** [comment]
@sircharlo, it's been a week. I think this is all the votes we're gonna get. Should we give Crowdin a go and see if we can [register as an open source project](https://crowdin.com/page/open-source-project-setup-request) to remove the limitations? If so, I suggest starting after the next release to prevent any last-minute problems. I figured out the correct crowdin configuration while I was testing it out, so I'll see if I can retrace it.

**Reply by sircharlo** [reply]
Alright! I'll register

**Reply by mtdvlpr** [reply]
I have a correct configuration at my forked repo:
https://github.com/mtdvlpr/meeting-media-manager/tree/master-crowdin

**Reply by sircharlo** [reply]
Configuration created; those who have created a CrowdIn login should have received an invite. Those who didn't let me know and Ill add you as need be.

**Reply by mtdvlpr** [reply]
Nice! The docs branch hasn't been added yet (see [docs-crowdin branch](https://github.com/mtdvlpr/meeting-media-manager/tree/docs-crowdin). If I could be a manager I can add it myself :)

**Reply by mtdvlpr** [reply]
Or are you waiting for the open source to go through? Since I think the docs branch will immediately exceed the free limit of 60.000 words...

**Reply by sircharlo** [reply]
Open source already went through; however, we're still only allowed one branch maximum from what I can see (trying to add the docs branch immediately results in an error.)

**Reply by mtdvlpr** [reply]
Ah, okay. Mind if I give it a go?

**Reply by mtdvlpr** [reply]
@sircharlo, I'd need manager rights...

**Reply by sircharlo** [reply]
Can't with the open source plan....

**Reply by mtdvlpr** [reply]
Oof, okay. The docs branch should be configured like so:
```
files:
  - source: /_data/en.json
    translation: /_data/%two_letters_code%.json
  - source: /locales/en/index.md
    translation: /locales/%two_letters_code%/index.md
  - source: /locales/en/_posts/*
    translation: /locales/%two_letters_code%/_posts/%original_file_name%
```

With the language mappers for Portuguese and Nigerian Pidgin overwriting the two letters code.

**Reply by mtdvlpr** [reply]
And duplicates handling set to `hide (regular)`

**Reply by mtdvlpr** [reply]
@sircharlo, it seems you're still on the Free plan (which allows 1 branch and no extra managers.
<details><summary>See image</summary>

![Screenshot_20230109_081130_Gmail](https://user-images.githubusercontent.com/46671786/211256776-f1205671-0dc0-4182-8ae4-6294e07288bb.png)

</details>

**Reply by mtdvlpr** [reply]
@sircharlo, it seems, because you made the integration, only you can change it (add the docs branch).

**Reply by sircharlo** [reply]
I was able to add you as manager @mtdvlpr but still getting an error with the GitHub integration when adding more than one branch. I've reached out to the onboarding team  so that they can look into it.

**Reply by sircharlo** [reply]
Couldn't we just run the docs site from the master branch but the docs subfolder? To avoid all these complications.

**Reply by mtdvlpr** [reply]
Yeah, if all it adds is complication, we'd best revert to one branch. I'll make a PR for it :)

**Reply by sircharlo** [reply]
The support team just got back to me. They manually increased the branch limit. It works now!

**Reply by mtdvlpr** [reply]
Then I suggest keeping it as is.

**Comment by mtdvlpr** [comment]
@sircharlo, I found 2 useful Crowdin apps to add to the Crowdin project:
- https://crowdin.com/store/apps/spellmanager
- https://crowdin.com/store/apps/duplicates

Only you can add them as tools. Be sure to select `all project members` for who can use the tool (if you decide to add them)

**Reply by mtdvlpr** [reply]
@sircharlo, found another one :)
https://crowdin.com/store/apps/extractor

**Reply by mtdvlpr** [reply]
@sircharlo, with this one we can show progress per language again:
https://crowdin.com/store/apps/bds

**Comment by comradekingu** [comment]
Consider Weblate, which having used all of those extensively seems an obvious choice.
Crowdin has egregious terms and fails as a tool because it is not possible to navigate strings easily, see more languages at once, or even get changes past a voting system that values first-movers higher by default for whatever reason.
